CMakeLists.txt

```diff
@@ -49,22 +49,37 @@ set(SOURCE
 		src/RCoreMutex.h
 		src/RCoreMutex.cpp)
 
+set(ASM_SOURCE
+		src/SleighAsm.h
+		src/SleighAsm.cpp
+		src/asm_ghidra.cpp)
+
 find_package(Radare2 REQUIRED)
 
 if(BUILD_CUTTER_PLUGIN)
 	add_subdirectory(cutter-plugin)
 endif()
 
-add_library(core_ghidra SHARED ${SOURCE})
+add_library(core_ghidra SHARED ${CORE_SOURCE})
 target_link_libraries(core_ghidra ghidra_decompiler_base ghidra_libdecomp ghidra_decompiler_sleigh)
 target_link_libraries(core_ghidra pugixml)
 target_link_libraries(core_ghidra Radare2::libr)
 set_target_properties(core_ghidra PROPERTIES
 		OUTPUT_NAME core_ghidra
 		PREFIX "")
 
+add_library(asm_ghidra SHARED ${ASM_SOURCE})
```

Comment:
We should have an compile time option to enable the sleigh integration that is disabled by default while it is still being actively developed. Put the following on top, where the other `option()`s are, then use `if(...)` to switch:
```
option(BUILD_SLEIGH_PLUGIN "Build asm plugin to use sleigh as a disassembler in r2" OFF)
```
If you have a better idea for the option name, feel free to suggest it.